website/docs/language/settings/backends/s3.mdx

```diff
@@ -224,6 +224,41 @@ terraform {
 }
 ```
 
+#### Assume Role With Web Identity Configuration
+
+The following `assume_role_with_web_identity` configuration block is optional:
+
+* `role_arn` - (Required) Amazon Resource Name (ARN) of the IAM Role to assume.
+Can also be set with the `AWS_ROLE_ARN` environment variable.
+* `duration` - (Optional) The duration individual credentials will be valid.
+Credentials are automatically renewed up to the maximum defined by the AWS account.
+Specified using the format `<hours>h<minutes>m<seconds>s` with any unit being optional.
+For example, an hour and a half can be specified as `1h30m` or `90m`.
+Must be between 15 minutes (15m) and 12 hours (12h).
+* `policy` - (Optional) IAM Policy JSON describing further restricting permissions for the IAM Role being assumed.
```

Comment:
Would it be possible for you to provide an example of using this?

E.g. constraining the policy to only create s3 buckets.

Reply:
@cube2222 Added an example of using a policy. WDYT?